### PR TITLE
Switch from jest to vitest

### PR DESCRIPTION
**Pull request by silverwind**
Even if we are not bundling with `vite` yet, we can use `vitest` in place of Jest which brings a few benefits like not requiring to use `NODE_OPTIONS` to run and having sane module resolution.

It's possible to also use `jest-extended` with vitest, but I opted to not do so for now because it brings heavyweight dependencies and it was trivial to just rewrite the affected matchers to be compatible.

This PR also removes 153 JS dependencies, which is certainly nice.

### CODE REVIEW

**Review by 6543 — APPROVED** [review]

**Review by wxiaoguang — APPROVED** [review]

File: vitest.config.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import {dataToEsm} from '@rollup/pluginutils';
+import {extname} from 'path';
+
+function stringPlugin() {
+  return {
+    name: 'string-plugin',
+    enforce: 'pre',
+    async load(id) {
+      const path = id.split('?')[0];
+      if (!['.svg'].includes(extname(path))) return null;
```
Comment (silverwind) [inline vitest.config.js]:
```suggestion
      if (extname(path) !== '.svg') return null;
```

**Comment by KN4CK3R** [comment]
> This PR also removes 153 JS dependencies, which is certainly nice.

npm install:
https://img-9gag-fun.9cache.com/photo/aQEpZB7_460svvp9.webm

**Comment by silverwind** [comment]
> > This PR also removes 153 JS dependencies, which is certainly nice.
> 
> npm install: https://img-9gag-fun.9cache.com/photo/aQEpZB7_460svvp9.webm

image

**Comment by silverwind** [comment]
[`threads`](https://vitest.dev/config/#threads) option is enabled again. It makes local tests a second slower but provides isolation between tests and it should overall become faster with increased number of tests.

**Review by 6543 — APPROVED** [review]

**Pull request merged**